src/main/java/org/snomed/snowstorm/core/data/services/ConceptUpdateHelper.java

```diff
@@ -582,6 +581,17 @@ void doDeleteMembersWhereReferencedComponentDeleted(Set<String> entityVersionsDe
 		// Mark updates
 		for (C newComponent : newComponents) {
 			final C existingComponent = existingComponentMap.get(newComponent.getId());
+
+			// Trying Concept module in attempt to restore effective time for the case
+			// where content has changed and then been reverted.
+			if (defaultModuleId != null) {
+				newComponent.setModuleId(newConcept.getModuleId());
+				newComponent.updateEffectiveTime();
+				if (newComponent.getEffectiveTime() == null) {
+					newComponent.setModuleId(defaultModuleId);
+				}
+			}
+
```

Comment:
I added in the null-check as not all branches have a defaultModuleId. Without this check, a NullPointerException was eventually being thrown in various tests.

Reply:
As discussed, the check for defaultModuleId here means that this code will only apply to MS branches so that avoids confusion with the Core / Model Module Ids.   It would be worth thinking about MS customers who might work in more than one module as we won't know which one to reset to.   I'll ask Andrew about that.

So you're saying here:    all components should take the moduleId of the parent concept, unless they've been updated in which case they should take the default module id.    I think I'm concerned about a case where a  model module concept has a description in the core module (which is wrong), but this code will automatically correct that which will make it look new and then pull the description over to the MS module.   I'll see if I can find an example of that for you.